go/pkg/log/server/property_test.go

```diff
@@ -168,19 +168,32 @@ func compareModelLogRecordToLogRecord(t *rapid.T, modelLogRecord ModelLogRecord,
 // is the same in both the model and the SUT
 func (suite *LogServerTestSuite) invariantLogsAreTheSame(ctx context.Context, t *rapid.T) {
 	for id, model_log := range suite.model.CollectionData {
-		pulled_log, err := suite.lr.PullRecords(ctx, id.String(), 0, len(model_log), time.Now().UnixNano())
+		next_offset := suite.model.CollectionCompactionOffset[id] + 1
+		pulled_log, err := suite.lr.PullRecords(ctx, id.String(), int64(next_offset), len(model_log), time.Now().UnixNano())
```

Comment:
I think this changes the behavior of this invariant, before - this invariant scanned the entire log and made sure that even after purges its the same. However, now we will only scan from the compaction offset onwards.

Should we instead scan from the purge offset onwards, so that the entire log that is still valid is scanned? 

This will inadvertently check the this invariant too, but we can also add a new check for that to detect any bugs with pulling at a specific offset.  

"invariantLogsAreTheSame" vs "invariantPulledLogsAfterCompactionAreTheSame"

Reply:
good point and observation